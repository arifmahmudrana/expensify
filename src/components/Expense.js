import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const Expense = ({description, amount, createdAt, id}) => (
  <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <p>{numeral(amount/100).format('$0,0.00')} - {moment(createdAt).format('Do MMM, YYYY')}</p>
  </div>
);

export default Expense;