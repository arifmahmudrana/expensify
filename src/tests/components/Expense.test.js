import React from 'react';
import { shallow } from 'enzyme';
import Expense from '../../components/Expense';
const expense = {
  id: '1',
  description: 'Description 1',
  amount: 100,
  note: 'Note 1',
  createdAt: 1541604692290
};

test('expense should render correctly', () => {
  const wrapper = shallow(<Expense {...expense} />);

  expect(wrapper).toMatchSnapshot();
});