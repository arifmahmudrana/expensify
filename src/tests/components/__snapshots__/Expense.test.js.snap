// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expense should render correctly 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      Description 1
    </Link>
  </h3>
  <p>
    $1.00
     - 
    7th Nov, 2018
  </p>
</div>
`;
